cylc/flow/config.py

```diff
@@ -399,7 +399,7 @@ def __init__(
                 )
 
         if (self.cfg['scheduling']['final cycle point'] is not None and
-                self.cfg['scheduling']['final cycle point'].strip() is ""):
+                self.cfg['scheduling']['final cycle point'].strip() == ""):
```

Comment:
For an empty string, shouldn't we just do `if not blah:`?

Reply:
(Interestingly `is` works fine, so all empty strings reference the same object).

Reply:
(So the original code is not actually wrong, strictly speaking)

Reply:
Already in bed. I read your comment and had to double check it.

The if us actually for the empty string. If it were for not none and not "" then I think we could kill both by not only.

I've recompiled python, the pydebug is a `./configure` flag. Raised other warning that I'll check tomorrow.

Reply:
>(So the original code is not actually wrong, strictly speaking)

Yup. Not a bug. Just removes a warning (hidden in normal python installations)

Reply:
I meant:
```python
if (self.cfg['scheduling']['final cycle point'] is not None and
    not self.cfg['scheduling']['final cycle point'].strip()):
```
(that's correct isn't it?)

Reply:
Oh, true! You are correct. Already in bed, so can edit the commit tomorrow (or feel free to edit this branch/commit)

Reply:
(and I deserve a big d'oh! haha)

Reply:
S'alright - it's usually me going d'oh these days :grin:  ... :sleeping: for me now too.

Reply:
Change updated! Hopefully with no bloopers this time (I re-read that `if` statement a few times to make sure I got it right haha)